lib/spack/spack/cmd/uninstall.py

```diff
@@ -344,23 +257,9 @@ def get_uninstall_list(args, specs, env):
     # Compute the set of specs that should be removed from the current env.
     # This may overlap (some specs may be uninstalled and also removed from
     # the current environment).
-    if args.remove:
-        remove_specs = set(base_uninstall_specs)
-        if args.dependents:
-            # Any spec matched from the cli, or dependent of, should be removed
-            # from the environment
-            for spec, lst in active_dpts.items():
-                remove_specs.update(lst)
-    else:
-        remove_specs = set()
-
-    all_uninstall_specs -= remove_only
-    # Inefficient topological sort: uninstall dependents before dependencies
-    all_uninstall_specs = sorted(
-        all_uninstall_specs, key=lambda x: sum(1 for i in x.traverse()), reverse=True
-    )
+    remove_specs = all_specs_in_env(env, all_uninstall_specs) if args.remove else []
 
-    return list(all_uninstall_specs), list(remove_specs)
+    return list(frozenset(all_uninstall_specs) - frozenset(remove_only)), remove_specs
```

Comment:
Why are we using a frozenset instead of a set?

Reply:
I think when I wrote this I mistakenly thought frozenset had better performance, but that is a lie.